### PR TITLE
Update StartNodeMaintenanceModal to use useK8sWatchResource hook

### PR DESCRIPTION
**Pull request by jtomasek**
... instead of Firehose wrapper

### CODE REVIEW

File: frontend/packages/metal3-plugin/src/components/modals/StartNodeMaintenanceModal.tsx [hunk 1]
```diff
@@ -20,11 +11,25 @@ import {
...
-import { RootState } from '@console/internal/redux';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+const cephClusterResource = {
+  kind: referenceForModel(CephClusterModel),
+  namespaced: false,
+  isList: true,
+};
+
+export type StartNodeMaintenanceModalProps = {
```
Comment (rawagner) [inline frontend/packages/metal3-plugin/src/components/modals/StartNodeMaintenanceModal.tsx:22]: you could reuse `HandlePromiseProps` and `ModalComponentProps`

```suggestion
export type StartNodeMaintenanceModalProps = HandlePromiseProps & ModalComponentProps & {
  nodeName: string;
```

**Comment by rawagner** [comment]
:+1:

nice

**Comment by spadgett** [comment]
/hold
for #5100

**Pull request merged**